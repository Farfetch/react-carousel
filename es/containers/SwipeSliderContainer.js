function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

import { CarouselContext } from "../context";
import { SwipeSlider } from "../components";
import React from 'react';

var SwipeSliderContainer = function SwipeSliderContainer(props) {
  return /*#__PURE__*/React.createElement(CarouselContext.Consumer, null, function (_ref) {
    var activeItem = _ref.activeItem,
        isRTL = _ref.isRTL,
        itemsToShow = _ref.itemsToShow,
        isInfinite = _ref.isInfinite,
        goNext = _ref.goNext,
        goPrev = _ref.goPrev,
        setItemsLength = _ref.setItemsLength,
        setIsMovementBlocked = _ref.setIsMovementBlocked,
        isMovementBlocked = _ref.isMovementBlocked,
        direction = _ref.direction;
    return /*#__PURE__*/React.createElement(SwipeSlider, _extends({}, props, {
      activeItem: activeItem,
      isRTL: isRTL,
      itemsToShow: itemsToShow,
      isInfinite: isInfinite,
      goNext: goNext,
      goPrev: goPrev,
      setItemsLength: setItemsLength,
      setIsMovementBlocked: setIsMovementBlocked,
      isMovementBlocked: isMovementBlocked,
      direction: direction
    }));
  });
};

export default SwipeSliderContainer;