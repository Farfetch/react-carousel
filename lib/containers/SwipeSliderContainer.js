"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _context = require("../context");

var _components = require("../components");

var _react = _interopRequireDefault(require("react"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

var SwipeSliderContainer = function SwipeSliderContainer(props) {
  return /*#__PURE__*/_react["default"].createElement(_context.CarouselContext.Consumer, null, function (_ref) {
    var activeItem = _ref.activeItem,
        isRTL = _ref.isRTL,
        itemsToShow = _ref.itemsToShow,
        isInfinite = _ref.isInfinite,
        goNext = _ref.goNext,
        goPrev = _ref.goPrev,
        setItemsLength = _ref.setItemsLength,
        setIsMovementBlocked = _ref.setIsMovementBlocked,
        isMovementBlocked = _ref.isMovementBlocked,
        direction = _ref.direction;
    return /*#__PURE__*/_react["default"].createElement(_components.SwipeSlider, _extends({}, props, {
      activeItem: activeItem,
      isRTL: isRTL,
      itemsToShow: itemsToShow,
      isInfinite: isInfinite,
      goNext: goNext,
      goPrev: goPrev,
      setItemsLength: setItemsLength,
      setIsMovementBlocked: setIsMovementBlocked,
      isMovementBlocked: isMovementBlocked,
      direction: direction
    }));
  });
};

var _default = SwipeSliderContainer;
exports["default"] = _default;
module.exports = exports.default;