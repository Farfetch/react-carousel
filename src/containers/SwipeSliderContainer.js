import { CarouselContext } from '#context';
import { SwipeSlider } from '#components';
import React from 'react';

const SwipeSliderContainer = (props) => (
    <CarouselContext.Consumer>
        {({
            activeItem,
            isRTL,
            itemsToShow,
            isInfinite,
            goNext,
            goPrev,
            setItemsLength,
            setIsMovementBlocked,
            isMovementBlocked,
            direction,
            containerRef,
        }) => (
            <SwipeSlider
                {...props}
                activeItem={activeItem}
                isRTL={isRTL}
                itemsToShow={itemsToShow}
                isInfinite={isInfinite}
                goNext={goNext}
                goPrev={goPrev}
                setItemsLength={setItemsLength}
                setIsMovementBlocked={setIsMovementBlocked}
                isMovementBlocked={isMovementBlocked}
                direction={direction}
                ref={containerRef}
            />
        )}
    </CarouselContext.Consumer>
);

export default SwipeSliderContainer;
