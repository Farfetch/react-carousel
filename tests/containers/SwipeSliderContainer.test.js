import { createSerializer } from 'enzyme-to-json';
import { shallow } from 'enzyme';
import React from 'react';
import SwipeSliderContainer from '../../src/containers/SwipeSliderContainer';

const mockContextCarousel = {
    activeItem: 1,
    isRTL: false,
    itemsToShow: 2,
    isInfinite: false,
    setItemsLength: jest.fn(),
    setIsMovementBlocked: jest.fn(),
    goNext: jest.fn(),
    goPrev: jest.fn(),
};

const mockChildren = <div />;

beforeAll(() => {
    expect.addSnapshotSerializer(createSerializer());
});

describe('<SwipeSliderContainer/>', () => {
    it('should render correctly', () => {
        const tree = shallow(
            <SwipeSliderContainer>{mockChildren}</SwipeSliderContainer>
        ).renderProp('children')({
            ...mockContextCarousel,
        });

        expect(tree).toMatchSnapshot();
    });

    it('should render correctly with custom Props', () => {
        const tree = shallow(
            <SwipeSliderContainer data-testId="test">
                {mockChildren}
            </SwipeSliderContainer>
        ).renderProp('children')({
            ...mockContextCarousel,
        });

        expect(tree.find('BaseSwipeSlider').prop('data-testId')).toBe('test');
    });

    it('should render correctly with correct Context', () => {
        const tree = shallow(
            <SwipeSliderContainer>{mockChildren}</SwipeSliderContainer>
        ).renderProp('children')({
            ...mockContextCarousel,
        });

        expect(tree.find('BaseSwipeSlider').prop('activeItem')).toBe(
            mockContextCarousel.activeItem
        );
        expect(tree.find('BaseSwipeSlider').prop('isRTL')).toBe(
            mockContextCarousel.isRTL
        );
        expect(tree.find('BaseSwipeSlider').prop('isInfinite')).toBe(
            mockContextCarousel.isInfinite
        );
        expect(tree.find('BaseSwipeSlider').prop('itemsToShow')).toBe(
            mockContextCarousel.itemsToShow
        );
        expect(tree.find('BaseSwipeSlider').prop('setItemsLength')).toBe(
            mockContextCarousel.setItemsLength
        );
        expect(tree.find('BaseSwipeSlider').prop('goNext')).toBe(
            mockContextCarousel.goNext
        );
        expect(tree.find('BaseSwipeSlider').prop('goPrev')).toBe(
            mockContextCarousel.goPrev
        );
    });
});
