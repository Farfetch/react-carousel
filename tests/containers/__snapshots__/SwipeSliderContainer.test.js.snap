// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SwipeSliderContainer/> should render correctly 1`] = `
<SwipeSlider
  activeItem={1}
  disableSwipe={false}
  goNext={[MockFunction]}
  goPrev={[MockFunction]}
  hasKeysNavigation={true}
  isInfinite={false}
  isRTL={false}
  itemsToShow={2}
  setItemsLength={[MockFunction]}
>
  <div />
</SwipeSlider>
`;
